### PR TITLE
CRM-16555 - Make Payment form credit / debit card changing available …

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
…to backoffice

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
It seems the assignment of <div id="billing-payment-block"> would ideally be done in the BillingBlock.tpl - some of the forms seem to include slightly more - but to get this to work from backoffice event edit forms need to add this div & also include the PaymentBlock.tpl

**Comment by eileenmcnaughton** [comment]
Update to the previous - I got around that by adding a wrapper since the div is not always 'clean' & I didn't know if I could make it clean

File: templates/CRM/Event/Form/Registration/Register.tpl [hunk 1]
```diff
@@ -370,7 +356,7 @@
       }
     }
   }
-  {/literal}{/if}{literal}
+  {/literal}{literal}
```
Comment (colemanw) [inline templates/CRM/Event/Form/Registration/Register.tpl]: ?

Comment (eileenmcnaughton) [inline templates/CRM/Event/Form/Registration/Register.tpl]: Yeah - that looks wrong - but not the 'current problem'

**Comment by colemanw** [comment]
+1 for changing "payment_processor" to the *actual* field name.

**Comment by eileenmcnaughton** [comment]
So @colemanw  the fundamental problem I'm having if coming out of https://github.com/civicrm/civicrm-core/blob/4.6/templates/CRM/common/paymentBlock.tpl#L66-66

Ideally I think (thought) both front end and back end forms should use the same piece of code to reload the billing block on payment processor change. There are 2 difference between them

1) front end uses payment_processor & back end uses payment_processor_d
2) front end is radio & back end is select

I *thought* I had it under control with the renaming BUT the front end is still not working with the reload js

**Comment by eileenmcnaughton** [comment]
The secondary problem I've removed from this PR to make it less confusing (although once again I have consistency in my confusion :-).

I wanted to re-use some of the other js functions for sanities sake - e.g per https://github.com/eileenmcnaughton/civicrm-core/commit/064cb5a676fd83b959f8ecf356bb4a64a622ac00 but was hitting problems around closure etc.

**Comment by colemanw** [comment]
@eileenmcnaughton so when you say "the same code to reload the billing block" which code do you mean exactly? Is this swap something you've already done?

**Comment by eileenmcnaughton** [comment]
@colemanw  by same piece of code I meant same piece of js - which was what your patch hopefully makes work

**Comment by eileenmcnaughton** [comment]
@colemanw to clarify, in earlier versions backoffice forms were not accessing
{include file="CRM/common/paymentBlock.tpl"}
I tried to amend it to work with both front end & back end forms - but for me with this patch it works on back-end forms and NOT front-end forms - although I have renamed the front end field to be the same

**Comment by colemanw** [comment]
So on the backend form I'm getting `Uncaught ReferenceError: buildRecurBlock is not defined` in the console. Know anything about that?

**Comment by colemanw** [comment]
Ok got the js basically working but... It seems that the billingBlockWrapper.tpl is what gets fetched by js, but that seems wrong, because it includes the js used to load itself, which then gets re-loaded and re-executed every time the payment processor is switched. What was the motivation for adding it?

**Comment by colemanw** [comment]
Also if I include **Authorize** as a payment processor then it works ok on the back-end, but on front-end contribution pages I get **Fatal error: Call to a member function getPaymentFormFields() on a non-object in /CRM/Core/Payment/Form.php on line 212**

**Comment by eileenmcnaughton** [comment]
I've added a PR against this branch with some re-work on paypal pro & paypal express

There is still a bug in it

https://github.com/eileenmcnaughton/civicrm-core/pull/4

ReferenceError: toggleConfirmButton is not defined

**Comment by eileenmcnaughton** [comment]
As discussed with @colemanw  - merging on test success

**Pull request merged**